components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/UserStoreConfigService.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.identity.user.store.configuration;
+
+import org.wso2.carbon.identity.user.store.configuration.dto.UserStoreDTO;
+import org.wso2.carbon.identity.user.store.configuration.utils.IdentityUserStoreMgtException;
+import org.wso2.carbon.ndatasource.common.DataSourceException;
+
+import java.util.Set;
+
+/**
+ * This interface used to expose user store management functionality as an OSGi Service.
+ */
+
+public interface UserStoreConfigService {
+
+    /**
+     * Add a userStore {@link UserStoreDTO}.
+     * @param userStoreDTO {@link UserStoreDTO} to insert.
+     * @throws IdentityUserStoreMgtException
+     */
+    void addUserStore(UserStoreDTO userStoreDTO) throws IdentityUserStoreMgtException;
+
+    /**
+     * Update the state of the userStore {@link UserStoreDTO}
+     * @param userStoreDTO {@link UserStoreDTO} to update
+     * @param  isStateChange true, if the update is a user store state change
+     * @throws IdentityUserStoreMgtException throws if an error occurred while updating the userStore
+     */
+    void updateUserStore(UserStoreDTO userStoreDTO, boolean isStateChange) throws IdentityUserStoreMgtException;
+
+    /**
+     * Update the name of the userStore domain.
+     * @param previousDomainName the domain name to be replaced
+     * @param userStoreDTO {@link UserStoreDTO} to update
+     * @throws IdentityUserStoreMgtException throws if an error occurred while updating the domain name.
+     */
+    void updateUserStoreByDomainName(String previousDomainName, UserStoreDTO userStoreDTO)
+            throws IdentityUserStoreMgtException;
+
+    /**
+     * Delete a userStore by domain name
+     *
+     * @param domain userStore domain name
```

Comment:
The exception is missing. Please fix in the other places as well.

Reply:
This is fixed in b7cbf33103eff335e20d1da97b1c2ef2d9770e9e